instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt

```diff
@@ -0,0 +1,154 @@
+package org.oppia.android.instrumentation.player
+
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.UiDevice
+import com.google.common.truth.Truth.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByDesc
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByRes
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.findObjectByText
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.scrollRecyclerViewTextIntoView
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.startOppiaFromScratch
+import org.oppia.android.instrumentation.testing.EndToEndTestHelper.waitForRes
+
+/** Tests for Explorations. */
+class ExplorationPlayerTest {
+  private lateinit var device: UiDevice
+
+  @Before
+  fun setUp() {
+    // Initialize UiDevice instance
+    device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+    device.startOppiaFromScratch()
+  }
+
+  @Test
+  fun testExploration_prototypeExploration_toolbarTitle_isDisplayedSuccessfully() {
+    navigateToPrototypeExploration()
+    assertThat(device.findObjectByRes("exploration_toolbar_title")).isNotNull()
+  }
+
+  @Test
+  fun testProtoTypeExploration_answerAllInteractionsAndCompleteExploration() {
+    navigateToPrototypeExploration()
+    completePrototypeExploration()
+
+    // Assert Topic Completed.
+    scrollRecyclerViewTextIntoView("Chapter 1: Prototype Exploration")
+    val chapterCompletedTick = device.findObjectByText(
+      "Chapter 1: Prototype Exploration"
+    )?.parent?.children?.get(2)
+    assertThat(chapterCompletedTick).isNotNull()
+  }
+
+  @Test
+  fun testImageRegionSelectionInteraction_answerAllInteractionsAndCompleteExploration() {
```

Comment:
@BenHenning Shall I remove this test case as this requires internet and we will be using localhost for images once 2.3 is merged.

> We should be using the test lesson that you introduced into Oppia backend, but we can't since the download piece isn't finished. Given that, I think it's reasonable to file a TODO issue for enabling the image region selection test, and disabling it for now (we should still implement it, but it can be reenabled and fixed once we can download the correct content to test).

This is what you mentioned in the mail.

Reply:
can we also remove the test case used to check the exploration title, as it was only used to verify if the e2e tests are running in the 2.1 pr

Reply:
Deferring to Ben for adding TODO as images will load from the local host after merging the last PR of this project.

Reply:
Could we instead ``@Ignore`` this & add a TODO pointing to the issue tracking properly hooking up to the backend once downloads are supported? (If there isn't such an issue, please file one).

That way we don't lose the actual context of the test in the meantime.

Reply:
Done

Reply:
https://github.com/oppia/oppia-android/issues/3697

Reply:
Thanks!